drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -579,4 +576,28 @@ def introspect_custom_data_element_definitions(set_definition_identifier: false)
 
     cded_records
   end
+
+  # Helper for determining CustomDataElementDefinition attributes
+  def self.infer_cded_field_type(item_type)
+    case item_type
+    when 'STRING', 'TEXT', 'CHOICE', 'TIME_OF_DAY', 'OPEN_CHOICE'
+      'string'
+    when 'BOOLEAN'
+      'boolean'
+    when 'DATE'
+      'date'
+    when 'INTEGER'
+      'integer'
+    when 'CURRENCY'
+      'float'
+    else
+      raise "unable to determine cded type for #{item_type}"
+    end
+  end
+
+  # Helper for determining CustomDataElementDefinition attributes
+  def self.generate_cded_field_label(item)
+    label = item.readonly_text || item.brief_text || item.text || key.humanize
```

Comment:
can we have empty strings here? Maybe should check with:
```suggestion
    label = item.readonly_text.presence || item.brief_text.presence || item.text.presence || key.humanize
```